src/test/incremental/hashes/let_expressions.rs

```diff
@@ -134,7 +134,7 @@ pub fn add_ref_in_pattern() {
 
 #[cfg(not(cfail1))]
 #[rustc_clean(cfg="cfail2",
-    except="hir_owner_nodes,typeck,optimized_mir")]
+    except="hir_owner_nodes,typeck,optimized_mir,promoted_mir")]
```

Comment:
I have no idea why this happens. Any idea how to investigate?

Reply:
In terms of why, I would look into the difference in queries for `promoted_mir`.

In terms of correct, you can use -Zdump-mir and examine PromoteTemps stage for different revisions. The ref pattern does cause promotion so that does look correct. Both version could be correct if promoted_mir wasn't queried before, but is now.

Reply:
Ok. Filing this under "unintended bugfixes".

Reply:
After looking a little bit more. The rustc_clean & rustc_dirty assertion have following behaviour:

> if the node wasn't previously evaluated and now is (or vice versa), then the node isn't actually clean or dirty.

And assertion from the attribute succeeds. In that case, the old implementation didn't query the `promoted_mir` in the first revision, and logs do confirm it, but the new implementation does apparently query it (if test fails without changing it). I don't see immediately why, It would be nice to find that out.

Reply:
Resolution: this modification was not needed. The question is void.